src/cli.rs

```diff
@@ -59,83 +57,83 @@ impl Opts {
     }
 }
 
-#[derive(StructOpt, Debug)]
-#[structopt(rename_all = "kebab-case")]
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
 pub struct RootOpts {
     /// Read configuration from one or more files. Wildcard paths are supported.
     /// File format is detected from the file name.
     /// If zero files are specified the default config path
     /// `/etc/vector/vector.toml` will be targeted.
-    #[structopt(
+    #[clap(
         name = "config",
         short,
         long,
         env = "VECTOR_CONFIG",
-        use_delimiter(true)
+        use_value_delimiter(true)
```

Comment:
`use_delimiter` is deprecated in clap 3.

This is chiefly what motivated this PR, per #11442